src/sql/workbench/services/connection/browser/connectionWidget.ts

```diff
@@ -65,11 +65,16 @@ export class ConnectionWidget extends lifecycle.Disposable {
 	private _serverGroupDisplayString: string = localize('serverGroup', "Server group");
 	private _trueInputValue: string = localize('boolean.true', 'True');
 	private _falseInputValue: string = localize('boolean.false', 'False');
+	private _encryptStrictValue: string = localize('encrypt.strict.value', 'strict');
```

Comment:
in ADS core, we should not need to be aware of these options. why not just handle them like generic contributed options?

Reply:
I need to customize behavior for Encrypt=Strict specially and register on selection event ([need this behavior](https://github.com/microsoft/azuredatastudio/pull/21256/files#diff-6cbed09935af5348af95a7fe0c504191ac5e2ee347e0325d5d89d6237dd90698R309)), any recommendations where should I do that?

Reply:
that should be handled in STS.

Reply:
please add gif to show the behaviors.

Reply:
Updated!

Reply:
Yeah, we should really try to find a way to make this generic. Because other providers may want the same type of functionality too in addition to the risk/annoyance of having provider-specific logic in core. 

What about doing something like I did for resource deployment with the "dynamic enablement" feature?

https://github.com/Microsoft/azuredatastudio/issues/13464

The package.json would then have a property on the option definition like visible that would contain the option name it depends on for determining whether it's visible

```json
        {
          "specialValueType": null,
          "isIdentity": false,
          "name": "trustServerCertificate",
          "displayName": "%mssql.connectionOptions.trustServerCertificate.displayName%",
          "description": "%mssql.connectionOptions.trustServerCertificate.description%",
          "groupName": "%mssql.connectionOptions.groupName.security%",
          "valueType": "boolean",
          "defaultValue": "false",
          "objectType": null,
          "categoryValues": null,
          "isRequired": false,
          "isArray": false,
          "showOnConnectionDialog": true,
          "visible": {
               "dependentOption": "encrypt",
               "values": ["true", "false"]
          }
        }
```

And when the connection dialog opens it will do something similar to the [hookUpDynamicEnablement](https://github.com/microsoft/azuredatastudio/pull/13464/files#diff-4339a5b5f41f5c3072b22046c84468152d18b82b159b6797bf4ebf481556ac29R341) function to add the necessary callbacks.

Reply:
I like that suggestion! 

It could be scaled to username/password too, to render on SqlAuth, and other toggles that we have... 
(not here, but maybe in future?) I'll start with basic design and we can take it from there.

Reply:
for this particular case, you also need to change the value of the hidden option, right? if that is the case, you probably should disable it instead of hiding it, so that it is clear what value is used.

Reply:
Changing value is not necessary since driver will ignore it anyways, but I'm trying to set it to default.

Reply:
Code is updated, behavior is exactly as above gif.